omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_2/FulfillerDetailsResource2_2.java

```diff
@@ -0,0 +1,107 @@
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingSubResource;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+/**
+ * {@link Resource} for Order, supporting standard CRUD operations on fulfiller_comment and fulfiller_status
+ */
+@SubResource(parent = OrderResource2_2.class, path = "fulfillerdetails", supportedClass = FulfillerDetails.class, supportedOpenmrsVersions = {
+		 "2.2.*" })
+
+public class FulfillerDetailsResource2_2 extends DelegatingSubResource<FulfillerDetails, Order, OrderResource2_2> {
+	
+	@Override
+	public FulfillerDetails newDelegate() {
+		return new FulfillerDetails();
+	}
+	
+	@Override
+	public FulfillerDetails save(FulfillerDetails delegate) {
+		Context.getOrderService().updateOrderFulfillerStatus(delegate.getOrder(), delegate.getFulfillerStatus(), delegate.getFulfillerComment());
+		return delegate;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		DelegatingResourceDescription delegatingResourceDescription = new DelegatingResourceDescription();
+		delegatingResourceDescription.addProperty("fulfillerStatus");
+		delegatingResourceDescription.addProperty("fulfillerComment");
+		return delegatingResourceDescription;
+	}
+	
+	@Override
+	public Model getCREATEModel(Representation rep) {
+		return new ModelImpl()
+				.property("fulfillerComment", new StringProperty())
+				.property("fulfillerStatus", new EnumProperty(null));
+	}
+	
+	@Override
+	public Model getGETModel(Representation rep) {
+		return new ModelImpl()
+				.property("fulfillerComment", new StringProperty())
+				.property("fulfillerStatus", new EnumProperty(null));
+	}
+	
+	@Override
+	public Order getParent(FulfillerDetails instance) {
+		return instance.getOrder();
+	}
+	
+	@Override
+	public void setParent(FulfillerDetails instance, Order parent) {
+		instance.setOrder(parent);
+	}
+	
+	@Override
+	public PageableResult doGetAll(Order parent, RequestContext context) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+	
+	@Override
+	public FulfillerDetails getByUniqueId(String uniqueId) {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+	
+	@Override
+	protected void delete(FulfillerDetails delegate, String reason, RequestContext context)
+			throws ResponseException {
+		Context.getOrderService().updateOrderFulfillerStatus(delegate.getOrder(), null, null);
+	}
+	
+	@Override
+	public void purge(FulfillerDetails delegate, RequestContext context) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+	
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription d = new DelegatingResourceDescription();
```

Comment:
calling return getRepresentationDescription(null) here actually avoid duplication since the properties are the same

Reply:
well I left it as it is because I changed getRepresentationDescription to actually add properties that are necessary to pass the test cases

Reply:
Those props should not be added